zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/authorization/DbAuthorizationState.java

```diff
@@ -23,63 +24,82 @@ public class DbAuthorizationState implements AuthorizationState, MutableAuthoriz
-  private final DbString resourceKey;
+  private final DbLong ownerKey;
   private final DbString resourceType;
-  private final DbCompositeKey<DbString, DbString> resourceCompositeKey;
-  private final DbCompositeKey<
-          DbCompositeKey<DbString, DbString>, DbCompositeKey<DbString, DbString>>
-      ownerAndResourceCompositeKey;
-  // owner key + owner type + resource key + resource type -> permission
+  private final DbString permissionType;
+  private final DbCompositeKey<DbString, DbString> resourceTypeAndPermissionTypeCompositeKey;
+  private final DbCompositeKey<DbLong, DbCompositeKey<DbString, DbString>>
+      ownerKeyAndResourceTypeAndPermissionTypeCompositeKey;
+  // owner key + resource type + permission type -> permissions
+  private final ColumnFamily<
+          DbCompositeKey<DbLong, DbCompositeKey<DbString, DbString>>, PersistedPermissions>
+      permissionsByOwnerKeyResourceTypeAndPermissionTypeColumnFamily;
+
+  private final DbString resourceId;
+  private final DbCompositeKey<DbString, DbLong> resourceIdAndOwnerKeyCompositeKey;
+  // resource id + owner key -> owner key + resource type + permission type
   private final ColumnFamily<
-          DbCompositeKey<DbCompositeKey<DbString, DbString>, DbCompositeKey<DbString, DbString>>,
-          PersistedPermissions>
-      ownerAuthorizationColumnFamily;
+          DbCompositeKey<DbString, DbLong>,
+          DbCompositeKey<DbLong, DbCompositeKey<DbString, DbString>>>
+      authorizationKeyByResourceIdAndOwnerKeyColumnFamily;
+
+  private final DbString ownerType;
+  // owner key -> owner type
+  private final ColumnFamily<DbLong, DbString> ownerTypeByOwnerKeyColumnFamily;
 
   public DbAuthorizationState(
       final ZeebeDb<ZbColumnFamilies> zeebeDb, final TransactionContext transactionContext) {
-    ownerKey = new DbString();
-    ownerType = new DbString();
-    resourceKey = new DbString();
+    ownerKey = new DbLong();
     resourceType = new DbString();
-    resourceCompositeKey = new DbCompositeKey<>(resourceKey, resourceType);
-    ownerAndResourceCompositeKey =
-        new DbCompositeKey<>(new DbCompositeKey<>(ownerKey, ownerType), resourceCompositeKey);
+    permissionType = new DbString();
+    resourceTypeAndPermissionTypeCompositeKey = new DbCompositeKey<>(resourceType, permissionType);
+    ownerKeyAndResourceTypeAndPermissionTypeCompositeKey =
+        new DbCompositeKey<>(ownerKey, resourceTypeAndPermissionTypeCompositeKey);
 
-    ownerAuthorizationColumnFamily =
+    permissionsByOwnerKeyResourceTypeAndPermissionTypeColumnFamily =
         zeebeDb.createColumnFamily(
-            ZbColumnFamilies.AUTHORIZATIONS_BY_USERNAME_AND_PERMISSION,
+            ZbColumnFamilies.PERMISSIONS_BY_OWNER_KEY_RESOURCE_TYPE_AND_PERMISSION_TYPE,
             transactionContext,
-            ownerAndResourceCompositeKey,
+            ownerKeyAndResourceTypeAndPermissionTypeCompositeKey,
             persistedPermissions);
+
+    resourceId = new DbString();
```

Comment:
❓ We've had some discussions recently about id this is `id` or `key`, I'm not informed enough on the overall picture but I think this could maybe be misleading based on those conversations

Reply:
Id is fine here. Key really refers to a key generated by Zeebe. This would be a long value. Here a resourceId is a String, eg `bpmnProcessId:foo`.